src/chat/chatter.py

```diff
@@ -305,15 +314,16 @@ def update_rank(self):
                                                      tooltip_str)
         else:
             icon_str = "newplayer"
+        if self.rankItem.toolTip() == tooltip_str:  # EARLY OUT
+            return
         self.rankItem.setIcon(util.THEME.icon("chat/rank/{}.png".format(icon_str)))
         self.rankItem.setToolTip(tooltip_str)
 
     def updateGame(self):
-        self.update_status_tooltip()
-        self.update_status_icon()
+        self.update_status()
```

Comment:
No need to fold these two into one if early returning is the only reason.

Reply:
I don't understand the logic of that sentence.

Reply:
You combined these 2 functions into one, probably so that early returning skips the second function too. I don't think it's a valid reason, this could very well be done using return values or exceptions.